src/app/services/qualification-filter.service.ts

```
@@ -0,0 +1,39 @@
+import {Injectable} from '@angular/core';
+import {Employee} from "../models/Employee";
+import {Qualification} from "../models/Qualification";
+import {FilterService} from "./filter.service";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class QualificationFilterService extends FilterService {
+
+  constructor() {
+    super();
+  }
+
+  filterQualificationsInUse(employees: Employee[]): Qualification[] {
+    let uniqueQualifications: Qualification[] = [];
+    const ids: number[] = [];
+    employees.forEach(employee => {
+        let qualifications: Qualification[] = employee.qualifications ?? [];
+        qualifications.forEach(qualification => {
+          const id = qualification.id ?? -1;
```

the check is not necessary. write a "!" somewhere if the compiler is unhappy:
data without id means something very very bad happened, so failing fast and crashing is not the worst.
also, since this still gets transpiled to js, i think it would be fine anyway

---

Just curious... why do you want to remove the check? Does it worsen the performance terribly? Or does it cause problems?